src/lib/fauna-client.ts

```diff
@@ -57,57 +44,39 @@ export default class FaunaClient {
       typecheck: opts?.typecheck ?? undefined,
       secret: opts?.secret ?? this.secret,
     };
-    return new Promise((resolvePromise, rejectPromise) => {
-      let req: any;
-      const onResponse = (http2ResponseHeaders: any) => {
-        const status = http2ResponseHeaders[constants.HTTP2_HEADER_STATUS];
-        let responseData = "";
+    const url = new URL(this.endpoint);
+    url.pathname = "/query/1";
+    const res = await fetch(url, {
+      method: "POST",
+      headers: {
+        authorization: `Bearer ${secret ?? this.secret}`,
+        "x-fauna-source": "Fauna Shell",
+        ...(typecheck !== undefined && { "x-typecheck": typecheck.toString() }),
+        ...(format !== undefined && { "x-format": format }),
```

Comment:
the `x-query-timeout-ms` header should be sent here

Reply:
good call, looks like we also weren't correctly passing the timeout to the client so resolving that as well.

Reply:
oh cool, nice